### PR TITLE
Add `db.getMany(keys)`

### PR DESCRIPTION
**Pull request by vweevers**
Tested against `leveldown`, `memdown`, `level-js`, `deferred-leveldown`, `encoding-down`, (indirectly) `levelup`, `subleveldown`.

Ref https://github.com/Level/community/issues/101

### CODE REVIEW

File: abstract-leveldown.js [hunk 1]
```diff
@@ -90,6 +94,51 @@ AbstractLevelDOWN.prototype._get = function (key, options, callback) {
...
+      this._nextTick(callback, err)
+      return callback.promise
+    }
+
+    serialized[i] = this._serializeKey(key)
+  }
+
+  this._getMany(serialized, options, callback)
+  return callback.promise
+}
```
Comment (vweevers) [inline abstract-leveldown.js:136]: Note that here we're doing everything that levelup normally does, as an experiment.

File: abstract-leveldown.js [hunk 2]
```diff
@@ -315,9 +364,40 @@ AbstractLevelDOWN.prototype._checkValue = function (value) {
...
+      return false
+    }
+
+    // TODO: deferred-leveldown and levelup are inconsistent: the former allows
+    // operations on any db status, the latter only if 'opening' or 'open'. We
+    // should define the scope of the deferredOpen feature. For now, pick the
+    // levelup behavior. Ref https://github.com/Level/community/issues/58
+    if (db.supports.deferredOpen && db.status === 'opening') {
+      return false
+    }
```
Comment (vweevers) [inline abstract-leveldown.js:396]: Not pretty but it works

File: abstract-leveldown.js [hunk 3]
```diff
@@ -90,6 +94,51 @@ AbstractLevelDOWN.prototype._get = function (key, options, callback) {
...
+  callback = getCallback(options, callback)
+  callback = catering.fromCallback(callback)
+  options = getOptions(options)
+
+  if (maybeError(this, callback)) {
+    return callback.promise
+  }
+
+  if (!Array.isArray(keys)) {
+    this._nextTick(callback, new Error('getMany() requires an array argument'))
```
Comment (vweevers) [inline abstract-leveldown.js:107]: I copied this async behavior from `levelup#batch`. Can't remember why we're not throwing there. I opted to be consistent with `levelup` just to make a choice, in the grand scheme of things this is not important.

File: test/get-many-test.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+  })
+}
+
+exports.getMany = function (test, testCommon) {
+  test('test getMany() support is reflected in manifest', function (t) {
+    t.is(db.supports && db.supports.getMany, true)
+    t.end()
+  })
+
+  test('test simple getMany()', function (t) {
```
Comment (vweevers) [inline test/get-many-test.js:65]: I copied this from `get-test.js`

File: abstract-leveldown.js (hunk 1 above)
Comment (vweevers) [reply abstract-leveldown.js:136]: Which means that `levelup` only has to do [this](https://github.com/Level/levelup/blob/a837e5407f2da39924ed327a655521fefcb0a214/lib/levelup.js#L188-L190):

```js
LevelUP.prototype.getMany = function (keys, options, callback) {
  return this.db.getMany(keys, options, callback)
}
```

File: abstract-leveldown.js (hunk 3 above)
Comment (vweevers) [reply abstract-leveldown.js:107]: It makes sense when considering `deferred-leveldown`, would otherwise throw once the db is open.

**Comment by vweevers** [comment]
Added in the last commit (1f1ab97):

- `AbstractLevelDOWN.prototype.isOperational()` which returns true if the db accepts operations. Normally that means when `db.status` is `'open'`, and `deferred-leveldown` can override `_isOperational()` to also accept operations when `db.status` is `'opening'`.
- More tests

And removed:

- `test custom _serialize*` tests which were not that valuable and not monkeypatch-safe

File: abstract-leveldown.js (hunk 1 above)
Comment (ralphtheninja) [reply abstract-leveldown.js:136]: So later on we can move over the `catering` stuff to `abstract-leveldown` one by one.

**Review by ralphtheninja — APPROVED** [review]

**Pull request merged**